doc/src/contracts/multisig_prover.md

```diff
@@ -62,59 +112,112 @@ Prover-->>-Relayer: returns GetProofResponse
+11. Multisig emits event indicating the multisig session has been completed
+12. Relayer queries Prover for the proof, using the proof ID
+13. Prover queries Multisig for the multisig session, using the session ID
+14. Multisig replies with the multisig state, the list of collected signatures so far and the snapshot of participants.
+15. If the Multisig state is `Completed`, the Prover finalizes constructing the proof and returns the `GetProofResponse` struct which includes the proof itself and the data to be sent to the destination gateway. If the state is not completed, the Prover returns the `GetProofResponse` struct with the `status` field set to `Pending`.
+
+
+## Update and confirm WorkerSet graph
 
-#[derive(QueryResponses)]
-pub enum QueryMsg {
-    #[returns(GetProofResponse)]
-    GetProof { multisig_session_id: Uint64 },
-}
+```mermaid
+graph TD
 
-pub enum ProofStatus {
-    Pending,
-    Completed { execute_data: HexBinary }, // encoded data and proof sent to destination gateway
-}
+r[Relayer]
+subgraph Axelar
+b[Prover]
+v[Voting Verifier]
+m[Multisig]
+end
 
-pub struct GetProofResponse {
-    pub multisig_session_id: Uint64,
-    pub message_ids: Vec<String>,
-    pub data: Data,
-    pub status: ProofStatus,
-}
+r--UpdateWorkerSet-->b
+b--RegisterWorkerSet-->m
+r--ConfirmWorkerSet-->b
+b--IsWorkerSetConfirmed-->v
 ```
 
-## Events
 
-```Rust
-pub enum Event {
-    ProofUnderConstruction {
-        multisig_session_id: Uint64,
-    },
-    SnapshotRotated {
-        key_id: String,
-        snapshot: Snapshot,
-        pub_keys: HashMap<String, HexBinary>,
-    },
-}
+## Update and confirm WorkerSet sequence diagram
+
+```mermaid
+sequenceDiagram
+autonumber
+participant External Chain
+participant Relayer
+box LightYellow Axelar
+participant Prover
+participant Voting Verifier
+participant Multisig
+end
+actor Signers
+Relayer->>+Prover: ExecuteMsg::UpdateWorkerSet
+alt no WorkerSet stored
+  Prover->>Prover: save new WorkerSet
+  Prover->>+Multisig: ExecuteMsg::RegisterWorkerSet
+  Multisig-->>-Prover: returns Response
+  Prover-->>-Relayer: returns Response
+else existing WorkerSet stored
+  Prover->>Prover: save new WorkerSet as the next WorkerSet
+  Prover->>+Multisig: ExecuteMsg::StartSigningSession
+  Multisig-->>Signers: emit SigningStarted event
+  Multisig->>-Prover: reply with session ID
+  Prover-->>Relayer: emit ProofUnderConstruction event
+  loop Collect signatures
+	  Signers->>+Multisig: signature collection
+  end
+  Multisig-->>-Relayer: emit SigningCompleted event
+  Relayer->>+Prover: QueryMsg::GetProof
+  Prover->>+Multisig: QueryMsg::GetSigningSession
+  Multisig-->>-Prover: reply with status, current signatures vector and snapshot
+  Prover-->>-Relayer: returns GetProofResponse
+  Relayer-->>External Chain: send proof and data
+  External Chain-->>+Relayer: emit OperatorshipTransferred event
+  Relayer->>+Voting Verifier: ExecuteMsg::ConfirmWorkerSet
+  Voting Verifier-->>-Relayer: returns Response
```

Comment:
these `returns Response` lines are not necessary and can be omitted. Makes the diagram easier to read, and they don't really add much